lib/improver/weighted_blend.py

```diff
@@ -57,20 +218,35 @@ def __init__(self, coord, coord_adjust=None):
     def __repr__(self):
         """Represent the configured plugin instance as a string."""
         return (
-            '<BasicWeightedAverage: coord = {0:s}>').format(self.coord)
+            '<WeightedBlend: coord = {0:s}>').format(self.coord)
 
     def process(self, cube, weights=None):
-        """Calculate weighted mean across the chosen coord
+        """Calculate weighted blend across the chosen coord, for either
+           probabilistic or percentile data. If there is a percentile
+           coordinate on the cube, it will blend using the
+           PercentileBlendingAggregator but the percentile coordinate must
+           have at least two points.
 
         Args:
             cube : iris.cube.Cube
                    Cube to blend across the coord.
             weights: Optional list or np.array of weights
-                     or None (equivalent to equal weights)
+                     or None (equivalent to equal weights).
 
         Returns:
             result : iris.cube.Cube
-                     containing the weighted mean across the chosen coord
+                     containing the weighted blend across the chosen coord.
+
+        Raises:
+            ValueError 1 : first argument not a cube.
+            ValueError 2 : The coord is not a coord with the cube.
+            ValueError 3 : If there is a percentile coord it is not a
+                           dimension coord in the cube.
+            ValueError 4 : If there are more than one percentile coords
+                           in the cube.
+            ValueError 5 : weights shape do not match the dimension
+                           of the coord.
+            Warning : coord not a dimension within the cube.
 
```

Comment:
I think looking at the warning message below that this would be meaningful if it reflected the warning message.
i.e.
Warning : Trying to blend across a scalar coordinate with only one value